src/main/java/trackr/model/task/TaskDescriptor.java

```diff
@@ -0,0 +1,76 @@
+package trackr.model.task;
+
+import java.util.Optional;
+
+import trackr.commons.util.CollectionUtil;
+
+/**
+ * Stores the details of a task. Each non-empty field value will replace the corresponding field value of the task.
+ */
+public class TaskDescriptor {
+    private TaskName taskName;
+    private TaskDeadline taskDeadline;
+    private TaskStatus taskStatus;
+
+    public TaskDescriptor() {}
+
+    /**
+     * Copy constructor.
+     */
+    public TaskDescriptor(TaskDescriptor toCopy) {
+        setTaskName(toCopy.taskName);
+        setTaskDeadline(toCopy.taskDeadline);
+        setTaskStatus(toCopy.taskStatus);
+    }
+
+    /**
+     * Returns true if at least one field is edited.
+     */
+    public boolean isAnyFieldEdited() {
```

Comment:
Since this task descriptor is used for find task also, should we change the name of this function to something else?

Reply:
But in find task, there is another function to incorporate its own variables as well.